### PR TITLE
fix(TCOMP-2111): support beam native mapper in ProducerFinder

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2121

- based on Chris' UT

### CODE REVIEW

File: component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    public Iterator<Record> find(final String familyName, final String inputName, final int version,
+            final Map<String, String> configuration) {
+        final ComponentInstantiator instantiator = getInstantiator(familyName, inputName);
+        final Mapper mapper = findMapper(instantiator, version, configuration);
+        try {
+            final Input input = mapper.create();
+            return iterator(input);
+        } catch (Exception e) {
+            log.warn("Component Kit Mapper instantiation failed, trying to wrap native beam mapper...");
+            final Object delegate = Delegated.class.cast(mapper).getDelegate();
```
Comment (clesaec) [inline component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java:61]: Is it Mapper or Input which contains delegated PTransform ?
(And exception occurs in iterator(input) call, so; all change could have been put here ?)
BTW, i can't find where TCK treat PTransform connectors.

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java [hunk 2]
```diff
@@ -76,7 +91,7 @@ static class InputIterator implements Iterator<Object> {
 
         private Object nextObject;
 
-        private boolean init = false;
+        private boolean init;
```
Comment (clesaec) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java:94]: Why remove explicit initialize .

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java [hunk 3]
```diff
@@ -88,26 +103,26 @@ private static Object findNext(final Input input) {
 
         @Override
         public boolean hasNext() {
-            synchronized (this.input) {
+            synchronized (input) {
```
Comment (clesaec) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java:106]: Why remove "this." ? it help to read code.

**Review by clesaec — COMMENTED** [review]
Well done,
Only some comments / questions ;

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java (hunk 2 above)
Comment (undx) [reply component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java:94]: matter of taste... default value for `boolean` primitive is `false`.

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java (hunk 3 above)
Comment (undx) [reply component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/ProducerFinderImpl.java:106]: matter of taste also. As there's no ambiguity on the variable vs a member.

**Review by clesaec — APPROVED** [review]
On implicit vs explicit, my taste is to prefer explicit (init boolean explicitly, always use this. ...), i find it easier to read.
But, it's OK

File: component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java (hunk 1 above)
Comment (undx) [reply component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java:61]: That's the mapper.
Don't get your final comment...

Comment (clesaec) [reply component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java:61]: ok, i see, name of S3Input class on cloud-component get me wrong, it's indeed a mapper
```java
@PartitionMapper(name = "S3Input")
@Documentation("This component reads data from S3.")
public class S3Input extends PTransform<PBegin, PCollection<Record>> {
```

**Pull request merged**